sparse_strips/vello_hybrid/examples/text.rs

```diff
@@ -0,0 +1,249 @@
+                    let mut pass = encoder.begin_render_pass(&RenderPassDescriptor {
+                        label: Some("Render to Texture Pass"),
+                        color_attachments: &[Some(wgpu::RenderPassColorAttachment {
+                            view: &texture_view,
+                            resolve_target: None,
+                            ops: wgpu::Operations {
+                                load: wgpu::LoadOp::Clear(wgpu::Color::BLACK),
+                                store: wgpu::StoreOp::Store,
+                            },
+                        })],
+                        depth_stencil_attachment: None,
+                        occlusion_query_set: None,
+                        timestamp_writes: None,
+                    });
+                    self.renderers[surface.dev_id].as_mut().unwrap().render(
+                        &self.scene,
+                        &mut pass,
+                        &render_params,
+                    );
+                }
+
+                device_handle.queue.submit([encoder.finish()]);
+                surface_texture.present();
+
+                device_handle.device.poll(wgpu::Maintain::Poll);
+            }
+            _ => {}
+        }
+    }
+}
+
+fn draw_text(ctx: &mut Scene) {
+    let font = Font::new(Blob::new(Arc::new(ROBOTO_FONT)), 0);
+    let font_ref = {
+        let file_ref = FileRef::new(font.data.as_ref()).unwrap();
+        match file_ref {
+            FileRef::Font(f) => f,
+            FileRef::Collection(collection) => collection.get(font.index).unwrap(),
+        }
+    };
+    let axes = font_ref.axes();
+    let size = 52_f32;
+    let font_size = skrifa::instance::Size::new(size);
+    let variations: Vec<(&str, f32)> = vec![];
+    let var_loc = axes.location(variations.as_slice());
+    let charmap = font_ref.charmap();
+    let metrics = font_ref.metrics(font_size, &var_loc);
+    let line_height = metrics.ascent - metrics.descent + metrics.leading;
+    let glyph_metrics = font_ref.glyph_metrics(font_size, &var_loc);
+
+    let mut pen_x = 0_f32;
+    let mut pen_y = 0_f32;
+
+    let text = "Hello, world!";
+
+    let glyphs = text
+        .chars()
+        .filter_map(|ch| {
+            if ch == '\n' {
+                pen_y += line_height;
+                pen_x = 0.0;
+                return None;
+            }
+            let gid = charmap.map(ch).unwrap_or_default();
+            let advance = glyph_metrics.advance_width(gid).unwrap_or_default();
+            let x = pen_x;
+            pen_x += advance;
+            Some(Glyph {
+                id: gid.to_u32(),
+                x,
+                y: pen_y,
+            })
+        })
+        .collect::<Vec<_>>();
+
+    ctx.set_paint(palette::css::WHITE.into());
+    let transform = Affine::scale(2.0).then_translate((0., f64::from(size) * 2.0).into());
+    ctx.set_transform(transform);
+
+    // Fill the text
+    ctx.glyph_run(&font)
+        .normalized_coords(vec![])
```

Comment:
It could be fun to really demonstrate variable font support here. The one tricky thing is that we don't have one in-repo. https://github.com/linebender/xilem/tree/main/xilem/resources/fonts/roboto_flex would be possible here (100KB)

Reply:
I ran out of time this morning to finish this. I've put it on my list of things to do.